packages/callout/src/Callout.spec.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { cleanup, render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import Callout, {
+  colorSets,
+  headerIcons,
+  headerLabels,
+  CustomCallout,
+  CustomCalloutProps,
+  Variant,
+} from './Callout';
+
+const className = 'test-classname';
+const children = 'this is the callout content.';
+const title = 'this is the callout title';
+const colorSet = colorSets[Variant.Note];
+const headerIcon = headerIcons[Variant.Note];
+const headerLabel = headerLabels[Variant.Note];
+
+function renderCustomCallout(props: Partial<CustomCalloutProps> = {}) {
+  const defaultProps = {
+    children,
+    colorSet,
+    headerLabel,
+  };
+  return render(<CustomCallout {...defaultProps} {...props} />);
+}
+
+describe('packages/callout', () => {
+  describe('Callout component', () => {
+    test('renders each variant correctly', () => {
+      for (const variant of Object.values(Variant)) {
+        const calloutMarkup = renderToStaticMarkup(
+          <Callout className={className} variant={variant}>
+            {children}
+          </Callout>,
+        );
+
+        const customCalloutMarkup = renderToStaticMarkup(
+          <CustomCallout
+            className={className}
+            colorSet={colorSets[variant]}
+            headerIcon={headerIcons[variant]}
+            headerLabel={headerLabels[variant]}
+          >
+            {children}
+          </CustomCallout>,
+        );
+
+        expect(calloutMarkup).toEqual(customCalloutMarkup);
+      }
+    });
+  });
+
+  describe('CustomCallout component', () => {
+    afterEach(async () => {
+      await cleanup();
+    });
+
+    test(`renders class name in classList: "${className}"`, async () => {
+      let { container } = renderCustomCallout({});
+      expect(container.firstChild).not.toHaveClass(className);
+      await cleanup();
+
+      ({ container } = renderCustomCallout({ className }));
```

Comment:
Rather than mutating the value of container, there's a supported API that we use elsewhere for updating the props and re-rendering a component through the testing library:

```js
const { container } = render(<MyComponent someProp={true} />);

render(
  <MyComponent someProp={false} />,
  { container },
);
```

Reply:
I actually think you should be able to grab container from screen.
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library

Reply:
Actually, even better would be just to use the `rerender` method: https://testing-library.com/docs/react-testing-library/api#rerender